substrate/zombienet-sdk/tests/staking_next/mod.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use subxt::{dynamic, dynamic::Value, ext::scale_value::Composite, OnlineClient, PolkadotConfig};
+use subxt_signer::sr25519::dev;
+use zombienet_sdk::{NetworkConfig, NetworkConfigBuilder};
+
+mod tests;
+
+/// Sets `ValidatorCount` in staking pallet to 500.
+async fn set_validator_count(
+	ah_client: &OnlineClient<PolkadotConfig>,
+	validator_count: u32,
+) -> Result<(), anyhow::Error> {
+	let set_validator_count = dynamic::tx("Staking", "set_validator_count", vec![validator_count]);
+	let sudo = dynamic::tx("Sudo", "sudo", vec![set_validator_count.into_value()]);
+	let alice = dev::alice();
+
+	ah_client
+		.tx()
+		.sign_and_submit_then_watch_default(&sudo, &alice)
+		.await?
+		.wait_for_finalized_success()
+		.await?;
+	Ok(())
+}
+
+/// Sets `Mode` in `StakingNextAhClient` pallet to `Active`.
+async fn activate_ah_client(rc_client: &OnlineClient<PolkadotConfig>) -> Result<(), anyhow::Error> {
+	let mode_value = Value::variant("Active", Composite::unnamed(vec![]));
+	let set_validator_count = dynamic::tx("StakingNextAhClient", "set_mode", vec![mode_value]);
+	let sudo = dynamic::tx("Sudo", "sudo", vec![set_validator_count.into_value()]);
+	let alice = dev::alice();
+
+	rc_client
+		.tx()
+		.sign_and_submit_then_watch_default(&sudo, &alice)
+		.await?
+		.wait_for_finalized_success()
+		.await?;
+	Ok(())
+}
+
+/// Builds a dummy RC and AH networks with the test runtimes from pallet-staking-async
+async fn build_network_config() -> Result<NetworkConfig, anyhow::Error> {
+	let images = zombienet_sdk::environment::get_images_from_env();
+	log::info!("Using images: {images:?}");
+	NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			r
+				.with_chain("custom")
+				.with_chain_spec_path("rc.json") // TODO: how to autogenerate this?
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![("-lparachain=trace").into()])
+				.with_default_resources(|resources| {
+					resources.with_request_cpu(2).with_request_memory("2G")
+				})
+				.with_node(|n| n.with_name("alice"))
+				.with_node(|n| n.with_name("bob"))
+		})
+		.with_parachain(|p| {
+			p.with_id(1100)
+			.with_chain_spec_path("parachain.json") // TODO: how to autogenerate this?
```

Comment:
esp if you move it to `./substrate/frame/staking-next`, I think it is okay for now to say in the README.md:

please first run `build-and-run-zn.sh compile`, and tweak this bash script to have an option to just prepare the two chain-specs, and not run ZN.